chapters/en/slides/module4/module4_02.Rmd

```diff
@@ -0,0 +1,567 @@
+     alt.Y(alt.repeat('row'), type='nominal'))
+ .repeat(column=numerical_columns, row=categorical_columns))
+```
+
+Notes:
+In addition to repeating charts with numerical data only,
+we can use the same principles to explore the relationships
+between the categorical and the numerical columns in our dataset.
+
+Here,
+we spread the three categorical variables out along the columns of the plot
+to be able to answer questions regarding how the different categories compare,
+e.g. which genre has the longest running movies?
+
+The categories labeled "null" are the ones which are missing a value in that categorical column in the dtaframe,
+but still has values in the numiercal column that is plotted on the x-axis.
+
+We have also removed the "Title" item from the list of categorical columns,
+as well as dropped it from the dataframe,
+since it would become a giant plot if all thuosands titles were included without limiting the plot height.
+sort the columns according to their 
+
+So what is the answer to our question,
+which genre has the longest running movies?
+It seems like Westerns, Musicals, and Dramas have the longest run times
+and we can also see that for the other plots
+it is Dramatization and Historical Fcition
+as well as PG-13 an R rated movies tha run the longest.
+
+You might have noticed that it was a bit hard to compare the boxes because they were not sorted.
+Since Altair requires a list of names to sort boxplots,
+we would have needed to pass a list of all the names from all the categorical columns,
+which would have looked like this:
+
+```python
+running_time_order = []
+for groupby_col in ['Major Genre', 'Creative Type', 'MPAA Rating']:
+    running_time_order.extend(
+        movies_extended
+        .groupby(groupby_col)
+        .median()
+        ['Running Time min']
+        .sort_values()
+        .index
+        .to_list())
+...
+alt.Y(alt.repeat('row'), type='nominal', sort=running_time_order))
+...
+```
+
+TODO I will revisit to name dataframe columns something else since it gets messy here when we have plot columns and dataframe columns. Maybe "variabe" as Tiff originally suggested, but that can be ambiguous when there are variables outside the dataframe. Feature is a bit ambiguous too, the most specific might be "field", but that's probably also the lest used... might just go with 'variable'.
+
+---
+
+## Exploring the categorical columns helps us understand the data further
+
+```{python}
+movies_extended.describe(exclude='number')
+```
+
+Notes:
+Next we look into the categorical columns.
+As we saw in last chapter,
+we look at the counts of observations for the values in each of the columns.
+
+As above,
+we start by printing information about the most frequently occuring categorical values
+in each of the columns.
+
+We can already here see what the most commons values are for each column,
+but let's visualize these in a bar chart
+and crossreference this table with our plot on the next slide.
```

Comment:
It might be good to give some examples of why we might be interested in these counts. One example is that, if your goal was to build a machine learning classification model, then have an unbalanced data set for the thing you are trying to predict (i.e., uneven numbers of things in the various categories) then you are going to have to do some extra things in your analysis to deal with that. There are lot's of other examples in statistical analysis, where uneven categories can change how you need to do your downstream analysis. And if you miss this, your results will not be as good/accurate as they could be.